aidial_adapter_openai/utils/merge_chunks.py

```diff
@@ -0,0 +1,31 @@
+from typing import TypeVar
+
+from aidial_sdk.utils.merge_chunks import merge
+
+_Chunk = TypeVar("_Chunk", bound=dict)
+
+
+def merge_chunks(*chunks: _Chunk) -> _Chunk:
+    """
+    The recursive merging procedure that avoids merging top-level atomic fields
+    (e.g. "id", "created", "model", "object", "system_fingerprint") and
+    instead chooses an _override_ merging strategy for such fields.
+    Non-atomic fields (e.g. "choice", "usage") are merged following
+    the standard recursive merging procedure.
+    """
+
+    assert len(chunks) > 0, "At least one chunk must be provided"
+
+    target = chunks[0]
+
+    def _generator():
+        yield target
+        for source in chunks[1:]:
+            source = source.copy()
```

Comment:
It would be really cool to highlight for user, that we don't want to change actual source

```python
source_to_merge = source.copy()
...

del source_to_merge[key]
```

What do you think?

Reply:
Fixed via `source = chunk.copy()`